server/utils.py

```diff
@@ -326,3 +329,16 @@ def get_directory(user_id):
     if is_orcid_id(user_id):
         return "https://orcid.org"
     return "https://cilogon.org"
+
+
+def make_url_https(url):
+    """
+    Given an http url, return it as https
+
+    :param url: The http url
+    :type url: str
+    :return: The url as https
+    :rtype: str
+    """
+    res = url.replace("http", "https")
```

Comment:
This'll probably work 99% of the time but does odd/bad things with strings like:

```python
In [13]: "https://example.com".replace("http", "https")
Out[13]: 'httpss://example.com'
```

or 

```python
In [14]: "https://example.com/http-example".replace("http", "https")
Out[14]: 'httpss://example.com/https-example'
```

I think a more robust way to change a URI from HTTP to HTTPS would use urllib:

```python
import urllib

def make_url_https(url):
  parsed = urllib.parse.urlparse(url)
  return parsed._replace(scheme="https").geturl()
```

```python
In [26]: make_url_https("http://example.com/http-example")
Out[26]: 'https://example.com/http-example'

In [27]: make_url_https("http://example.com")
Out[27]: 'https://example.com'

In [28]: make_url_https("htts://example.com")
Out[28]: 'https://example.com'
```

What do you think?

Reply:
Made that test and also added those examples as unit tests